apps/services/notifications/src/app/modules/notifications/notifications.controller.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  Controller,
+  Post,
+  BadRequestException,
+  Req,
+  HttpCode,
+} from '@nestjs/common'
+import { ApiBody, ApiExtraModels, getSchemaPath } from '@nestjs/swagger'
+import { validate, ValidationError } from 'class-validator'
+import { Request } from 'express'
+import { NotificationProducerService } from './producer.service'
+import {
+  NewPostholfMessage,
+  Message,
+  TypeValidator,
+  ValidatorTypeMap,
+} from './dto/createNotification.dto'
+
+const throwIfError = (errors: ValidationError[]): void => {
+  if (errors.length > 0) {
+    throw new BadRequestException(
+      errors.reduce(
+        (acc, e) => acc.concat(Object.values(e.constraints ?? {})),
+        [] as string[],
+      ),
+    )
+  }
+}
+
+const validateMessage = async (body: Request['body']): Promise<Message> => {
+  const type = new TypeValidator()
+  Object.assign(type, body)
+  throwIfError(await validate(type, { forbidUnknownValues: false }))
+
+  const ValidatorClass = ValidatorTypeMap[type.type]
+  const message = new ValidatorClass()
+  Object.assign(message, body)
+  throwIfError(await validate(message, { forbidUnknownValues: true }))
+
+  return message
+}
+
+@Controller('notifications')
+@ApiExtraModels(NewPostholfMessage)
+export class NotificationsController {
+  constructor(
+    private readonly notificationProducer: NotificationProducerService,
+  ) {}
+
+  @Post()
+  @ApiBody({
+    schema: {
+      type: 'object',
+      oneOf: [{ $ref: getSchemaPath(NewPostholfMessage) }],
+    },
+  })
+  @HttpCode(202)
```

Comment:
```A 201 status code indicates that a request was successful and as a result, a resource has been created (for example a new page). The status code 202 indicates that server has received and understood the request, and that it has been accepted for processing, although it may not be processed immediately```

https://benramsey.com/blog/2008/04/http-status-201-created-vs-202-accepted/

I think 202 is a better fit, but I don't have a strong opinion on it

Reply:
I agree that 202 is a better fit, but I think it's better to keep to the docs wherever possible to maintain consistency.
Another way to think of this is that the entry was successfully created in the queue 🤷